src/main/kotlin/com/github/huntersherms/pojo2parquet/Pojo2Parquet.kt

```diff
@@ -0,0 +1,154 @@
+                .withCompressionCodec(GZIP)
+                .withWriteMode(ParquetFileWriter.Mode.OVERWRITE)
+                .build().use({ writer ->
+            pojos.forEach { writer.write(it) }
+        })
+        return file
+    }
+
+    /**
+     * Converts a parquet file to a list of Pojos
+     *
+     * Uses reflection so that providing an Avro schema is not necessary
+     */
+    fun parquet2Pojos(file: File): List<T> {
+
+        val pojos = mutableListOf<T>()
+        AvroParquetReader.builder<T>(Path(file.toURI()))
+                .withDataModel(ReflectData(clazz.classLoader))
+                .disableCompatibility() // always use this (since this is a new project)
+                .build().use({
+                    var pojo = it.read()
+                    while (null != pojo) {
+                        pojos.add(pojo)
+                        pojo = it.read()
+                    }
+        })
+        return pojos
+    }
+
+    /**
+     * Use in place of pojos2Parquet if your POJOs are Jackson annotated and you want the derived parquet column names
+     * to follow your annotations rather than your POJO property names.
+     */
+    fun jacksonAnnotatedPojos2Parquet(pojos: List<T>): File {
+
+        assert(pojos.isNotEmpty())
+
+        val file = File.createTempFile("parquet", ".gzip")
+
+        /**
+         * Constructs our Avro Mapper. For more infor on the added configuration see:
+         * https://github.com/FasterXML/jackson-dataformats-binary/issues/15
+         */
+        val mapper = AvroMapper(AvroFactory().enable(AvroGenerator.Feature.AVRO_FILE_OUTPUT))
+
+        //Derive our schema using Jackson
+        val schema = getAvroSchema(mapper)
+
+        val avroWriter = mapper.writer(AvroSchema(schema))
+
+        AvroParquetWriter.builder<GenericRecord>(Path(file.toURI()))
+                .withSchema(schema)
+                .withCompressionCodec(GZIP)
+                .withWriteMode(ParquetFileWriter.Mode.OVERWRITE)
+                .build().use({ writer ->
+                    pojos.forEach {
+                        DataFileReader.openReader(
+                                SeekableByteArrayInput(avroWriter.writeValueAsBytes(it)),
+                                GenericDatumReader<GenericRecord>(schema)).forEach {
+                            writer.write(it)
+                        }
+                    }
+                })
+        return file
+    }
+
+    /**
+     * Use in place of parquet2Pojos if your POJOs are Jackson annotated and you want Jackson to manage the mapping
+     * to your POJOs
+     */
+    fun parquet2JacksonAnnotatedPojos(file: File): List<T> {
+
+        val mapper = AvroMapper()
+
+        //Derive our schema using Jackson
+        val schema = getAvroSchema(mapper)
+
+        val avroReader = mapper.reader(AvroSchema(schema)).forType(clazz)
+
+        val pojos = mutableListOf<T>()
+
+		val datumWriter = GenericDatumWriter<GenericRecord>(schema)
```

Comment:
Bad indenting.

Reply:
weird... I think this is Github only? It is correctly indented on my end.

Reply:
Tabs vs spaces?